src/Deputy/DistrictLoader.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace AppBundle\Deputy;
+
+use AppBundle\Entity\District;
+use AppBundle\Geo\GeometryFactory;
+use Doctrine\Bundle\DoctrineBundle\Registry;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Serializer\Encoder\CsvEncoder;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+use Symfony\Component\Serializer\Serializer;
+
+class DistrictLoader
+{
+    private $em;
+    private $logger;
+    private $decoder;
+    private $doctrine;
+    private $geoCountries;
+    private $geoDistricts;
+    private $geometryFactory;
+
+    public function __construct(Registry $doctrine, GeometryFactory $geometryFactory, LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+        $this->doctrine = $doctrine;
+        $this->geometryFactory = $geometryFactory;
+        $this->em = $this->doctrine->getManager();
+        $this->decoder = new Serializer([new ObjectNormalizer()], [new JsonEncoder(), new CsvEncoder()]);
+    }
+
+    public function load(string $file, string $districtsFile, string $countriesFile): void
+    {
+        $districts = $this->decoder->decode(file_get_contents($file), 'csv', [CsvEncoder::DELIMITER_KEY => ';']);
+        $this->geoDistricts = $this->decoder->decode(file_get_contents($districtsFile), 'json')['features'];
+        $this->geoCountries = $this->decoder->decode(file_get_contents($countriesFile), 'json')['features'];
+
+        $this->batchInsertOrUpdateDistricts($districts);
+    }
+
+    public function batchInsertOrUpdateDistricts(array $districts): void
+    {
+        $this->logger->notice(sprintf('%s districts are about to be loaded', count($districts)));
+
+        $i = 0;
+
+        foreach ($districts as $district) {
+            $this->createOrUpdateDistrict($district);
+            $this->em->persist($this->createOrUpdateDistrict($district));
+            $this->em->flush();
+
+            if (0 === ++$i % 1000 || $i === count($districts)) {
+                $this->em->clear();
```

Comment:
`flush()` should be in that condition too no ?

Reply:
Not really, I want to flush after each insert/update.